similarity-data-impl/src/main/java/org/phenotips/data/similarity/internal/AbstractExome.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.phenotips.data.similarity.internal;
+
+import org.phenotips.data.similarity.Exome;
+import org.phenotips.data.similarity.Variant;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+
+/**
+ * Base class for implementing exome-related classes, specifically those that include scored genes with variants.
+ *
+ * @version $Id$
+ * @since 1.0M6
+ */
+public abstract class AbstractExome implements Exome
+{
+    /** The overall score for each gene with a variant. */
+    protected Map<String, Double> geneScores;
+
+    /** The variants in each gene. */
+    protected Map<String, List<Variant>> variants;
+
+    @Override
+    public Set<String> getGenes()
+    {
+        return Collections.unmodifiableSet(this.variants.keySet());
+    }
+
+    @Override
+    public Double getGeneScore(String gene)
+    {
+        return this.geneScores.get(gene);
```

Comment:
Good spot. Is the appropriate way of protecting against this to initialize `geneScores` in a constructor in the abstract class, or just check for null in the method (or both)?